plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/AbstractProductTemplate.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Product Block Editor abstract product template.
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplates;
+
+abstract class AbstractProductTemplate {
+
+    /**
+     * Block tree root.
+     */
+    const ROOT = 'root';
+
+    /**
+     * Index for the block properties.
+     */
+    const BLOCK_PROPERTIES_INDEX = 1;
+
+    /**
+     * Index for the child blocks.
+     */
+    const BLOCK_CHILDREN_INDEX = 2;
+
+    /**
+     * Property for the internal order.
+     */
+    const ORDER_PROPERTY = '_order';
+
+    /**
+     * The block template.
+     */
+    protected $template = array();
+
+    /**
+     * Cache of reference to inserted blocks location in tree.
+     */
+    protected $cache = array();
+
+    /**
+     * Add a field to the existing template.
+     */
+    public function add_field( $field ) {
+        $args   = wp_parse_args( $field, array( 'order' => 10 ) );
+        $id     = $args['id'] ?? null;
+        $block  = $args['block'];
+        $parent = $args['parent'];
+        $order  = $args['order'];
+
+        $block[ self::BLOCK_PROPERTIES_INDEX ][ self::ORDER_PROPERTY ] = $order;
+
+        $this->insert_block( $parent, $block, $id );
+    }
+
+    /**
+     * Insert a block into the template tree.
+     */
+    private function insert_block( $parent, $block, $id ) {
+        if ( $parent === self::ROOT ) {
```

Comment:
We don't currently guard against blocks being inserted at the root.  We could do so to try and enforce use of tabs/sections at the top level, but this is pretty easy to work around and might break some use cases so I think leaving it as is may be the best move, but open for discussion.

Reply:
I'd say to leave possible.